src/openfermion/transforms/_projection.py

```diff
@@ -0,0 +1,61 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+'''
+Functions to reduce the number of qubits involved
+in modelling a given system
+'''
+from openfermion.ops import QubitOperator
+import numpy
+
+
+def project_onto_sector(operator, qubits, sectors):
+    '''
+    Takes a QubitOperator, and projects out a list
+    of qubits, into either the +1 or -1 sector.
+    Note - this requires knowledge of which sector
+    we wish to project into.
+
+    Input:
+        operator: the QubitOperator to work on
+        qubits: a list of indices of qubits in
+           operator to remove
+        sectors: for each qubit, whether to project
+            into the 0 subspace (<Z>=1) or the
+            1 subspace (<Z>=-1).
+
+    Returns:
+        projected_operator: the resultant operator
+        error: the trace norm of the removed term.
+    '''
+    if type(operator) is not QubitOperator:
+        raise ValueError('''Input operator must be a QubitOperator''')
+
+    projected_operator = QubitOperator()
+    error = 0
+    for term,factor in operator.terms.items():
+
+        # Any term containing X or Y on the removed
+        # qubits has an expectation value of zero
+        if [t for t in term if t[0] in qubits
+                and t[1] in ['X', 'Y']]:
+            error += abs(factor)**2
+            continue
+
+        new_term = tuple((t[0]-len([q for q in qubits 
+                                    if q < t[0]]), t[1])
+                         for t in term if t[0] not in qubits)
+        new_factor =\
```

Comment:
How does the linter feel about this?

Reply:
The linter didn't complain, but it doesn't read that well. Rewritten.